beta/src/content/apis/react/memo.md

```diff
@@ -2,21 +2,314 @@
 title: React.memo
 ---
 
-<Wip>
+<Intro>
 
-This section is incomplete, please see the old docs for [React.memo.](https://reactjs.org/docs/react-api.html#reactmemo)
+`memo` lets you avoid re-rendering a component when its props are unchanged.
 
-</Wip>
+```
+export default memo(function Button({title}) {
+  return <button>{title}</button>
+});
+```
 
+</Intro>
 
-<Intro>
+- [Usage](#usage)
+  - [Avoiding re-rendering when props are unchanged](#avoiding-rerendering-when-props-are-unchanged)
+  - [Updating a memoized component using state](#updating-a-memoized-component-using-state)
+  - [Updating a memoized component using a context](#updating-a-memoized-component-using-state)
+  - [Minimizing props changes](#minimizng-props-changes)
+  - [Specifying a custom comparison function](#specifying-a-custom-comparison-function)
+- [Reference](#reference)
+- [Troubleshooting](#troubleshooting)
+  - [My component re-renders when a prop is an object or array](#my-component-rerenders-when-a-prop-is-an-object-or-array)
+  - [My component re-renders when a prop is a function](#my-component-rerenders-when-a-prop-is-a-function)
+
+---
+
+## Usage {/*usage*/}
+
+### Avoiding re-rendering when props are unchanged {/*avoiding-rerendering-when-props-are-unchanged*/}
+
+React normally re-renders a component whenever its parent re-renders. With `memo`, you can create a component that React will not re-render when its parent re-renders so long as its new props are the same as the old props. Such a component is said to be *memoized*.
+
+`memo` is a function: you use it by wrapping your component in a call to `memo` and using the value that it returns in place of your original component:
+
+```
+export default memo(function Greeting({name}) {
+  return <>Hello, {name}!</>;
+});
+```
+
+A React component should always render the same output if its props, state, and context haven't changed. By using `memo`, you are telling React that your component complies with this requirement, so React doesn't need to re-render as long as none of those things have changed. When you use `memo`, your component will still re-render if its own state changes or if a context that it's using changes.
+
+<Gotcha>
+
+Use `memo` only as a performance optimization, not to guarantee that a re-render
+will not happen. React may choose to re-render anyway in some situations, such as to save
+memory when a component is off-screen.
```

Comment:
I think the key idea we want to convey here is that if a component *breaks* due to a re-render, there's a bug you need to fix in the component. Usually it would be that component is impure, so we can link to `/learn/keeping-components-pure`.